### ISSUE TITLE
Proposition: Java import ordering

### ISSUE THREAD

**Issue by jmnarloch** [editorconfig/editorconfig#231]
I wasn't sure where to propose this idea so I open the issue here
At work we have this constant problems with formating Java code. Different project members have different IDE and formatting settings we had tried to tackle down by using unified formatting scheme. We introduced editorconfig and works fine with most of the code formatting except for the Java imports.

Could you please consider introducing some option of unifiing the Java import statements ordering that affterwards every plugin/IDE could implement in the same manner?

I think that afterwards other language could benefit from having the same kind of option/flag added in the future.

**Comment by jedmao** [comment]
EditorConfig doesn't deal with language-specific settings. You might want to look for a Java-specific tool to achieve your goal.

**Comment by hanswesterbeek** [comment]
Right, but the IntelliJ-editorconfig plugin make Intelli /does/ (inadvertentently?) cause the settings I provide for imports to be ignored.

**Comment by jkubrynski** [comment]
:+1: for introducing language specific features. Without it the project is almost useless

**Comment by tommyschnabel** [comment]
+1, would love to have this. It's just about once a week that someone at work has their imports order out of sync from our standard

**Comment by b-long** [comment]
Any movement on this?  Looks like the thread is fairly old.  Can the @editorconfig group weigh in?

**Comment by jedmao** [comment]
@b-long , the @editorconfig group [did weigh in](https://github.com/editorconfig/editorconfig/issues/231#issuecomment-139810471).

**Comment by holandaleo** [comment]
> EditorConfig doesn't deal with language-specific settings. 

I partially agree with that statement. EditorConfig already has a setting that is language-specific: continuation_indent_size. That setting will behave differently for each language. For example, java line continuation is based on semi-colon while XML line continuation is based on the angle-brackets and so on.
Language specifics could simply be ignored by editors that do not support a certain language.

**Comment by ppalaga** [comment]
Having every language specific rule to pass the voting here at the EditorConfig level is very impractical. What we need is an offcial EditorConfig support for property namespaces. Having that, we could found a separate community to define/maintain properties specific to java. Plz. upvote/comment https://github.com/editorconfig/editorconfig/issues/332 if you like the idea.

**Comment by jedmao** [comment]
@holandaleo `continuation_indent_size` is not a core setting.

**Comment by vikreinok** [comment]
In our team, most frequent merge conflict is caused due to incorrect java import  order order. 
We do appreciate 'wildcard_import_limit ', Still the order would be nice. 

Eclipse formatter plugin has it. 
For example config there looks like: java;javax;org;com;

**Comment by matthew16550** [comment]
An old issue but perhaps people still look here.

We can now use `ij_java_imports_layout` in `.editorconfig`:
```
ij_java_imports_layout = *,|,javax.**,java.**,|,$*
```
`|` means a blank line, `$*` means the static imports.

Also, wildcard imports can be avoided with the following.  But they will not work until IntelliJ 2021.1 ([IDEA-225733](https://youtrack.jetbrains.com/issue/IDEA-225733)).
```
ij_java_class_count_to_use_import_on_demand = 999
ij_java_names_count_to_use_import_on_demand = 999
```

**Comment by Lonzak** [comment]
That is nice for inteliJ but it doesn't work for other IDEs like eclipse. So this topic is still open imho.
